File: 1861-rotating-the-box/1861-rotating-the-box.py
class Solution:
    def rotateTheBox(self, boxGrid: List[List[str]]) -> List[List[str]]:
        m = len(boxGrid)
        n = len(boxGrid[0])
        res = [['.' for _ in range(m)] for _ in range(n)]
        for i in range(m):
            restPos = n - 1
            for j in range(n - 1, -1, -1):
                if boxGrid[i][j] == '#':
                    res[restPos][m - i - 1] = '#'
                    restPos -= 1
                elif boxGrid[i][j] == '*':
                    res[j][m - i - 1] = '*'
                    restPos = j - 1
        return res

File: 1861-rotating-the-box/1861-rotating-the-box.cpp
class Solution {
public:
    vector<vector<char>> rotateTheBox(vector<vector<char>>& box) {
        int m = box.size();
        int n = box[0].size();
        vector<vector<char>> res(n, vector<char>(m, '.'));
        for(int i = 0; i < m; i++)
        {
            int lastCorrectPlace = n - 1;
            for(int j = n - 1; j >= 0; j--)
            {
                if(box[i][j] == '#')
                {
                    res[lastCorrectPlace][m - i - 1] = '#';
                    lastCorrectPlace--;
                }
                else if(box[i][j] == '*')
                {
                    res[j][m - i - 1] = '*';
                    lastCorrectPlace = j - 1;
                }
            }
        }
        return res;
    }
};